src/size_hint.rs

```diff
@@ -74,6 +74,31 @@ pub fn mul_scalar(sh: SizeHint, x: usize) -> SizeHint {
     (low, hi)
 }
 
+/// Raise two correctly by a **SizeHint** exponent.
+#[inline]
+#[allow(dead_code)]
+pub fn two_exp(exp: SizeHint) -> SizeHint {
+    let (mut low, mut hi) = exp;
+
+
+    // saturating 'raise two by exponent'
+    let shl_range = 1_usize.leading_zeros() as usize;
+    low = if low <= shl_range {
+        1 << low
+    } else {
+        usize::MAX
+    };
```

Comment:
Isn't there `saturating_pow` or similar?

Reply:
Yes! I forgot to check if it `saturating_pow` had hit stable since the last time I worked on this.

Thinking about this... would it make sense to make this more general and add a `base` argument? e.g. `pow(base: usize, exp: SizeHint)`. Not sure about naming though if we consider a complementary function that raises a base `SizeHint` to an exponent might be needed in the future?

Of course adding this at all will depend on whether we think having a `size_hint' impl for `Powerset` is worth the costs.